src/main/java/com/vizor/unreal/util/Misc.java

```diff
@@ -246,6 +246,71 @@ else if (numBrackets < 0)
         }
     }
 
+    // s6fix @bernst - Generate C++ safe and compilable code.
+
+    /**
+     * Converts a string like "com.s6.core.image-service" to ComS6CoreImageService.
+     * Preserves case, except to upper case first letter to a word, where words are separated by underscores.
+     *
+     * In the future we can consider reworking the code to transform ssomething like "com.s6.core.image-service to "ComS6Core_ImageService".
+     * 
+     * @return Input "com.s6.core.image-service" to "ComS6CoreImageService"
+     */
+    public static String mixedCharacterStringToCppSafe(final String mixedCharacterString)
```

Comment:
I think this would be higher value if we split it into two function: `snakeCaseToPascalCase` and `cppSafeNameString` or `cppSafeIdentifierString` and then compose them where needed. This would allow us to do `snakeCaseToCamelCase(cppSafeIdentifierString(foo))` and other combinations as well.

Reply:
Their snakeCaseToCamelCase already supports PascalCase by setting firstLetterIsCapital to true.